Dockerfile

```diff
@@ -1,74 +1,137 @@
+    bash-completion \
+    # Provides command line and full screen utilities for browsing procfs, a "pseudo" file system dynamically generated by the kernel to provide information about the status 
+    # of entries in its process table (such as whether the process is running, stopped, or a "zombie"). Requires by IDEA to disable warnings. 
+    procps \
+    # Provides find utility. Fedora 32 doesn't contain find utility by default.
+    findutils \
+    # Check whether download Url represent IntelliJ IDEA to install required dependencies.
+    && if [ "${downloadUrl#*idea}" != "$downloadUrl" ]; then yum install -y java-11-openjdk-devel; else echo "Not IntelliJ Idea"; fi \
+    # Check whether download Url represent PyCharm to install required dependencies.
+    && if [ "${downloadUrl#*pycharm}" != "$downloadUrl" ]; then yum install -y python2 python3 python3-distutils-extra python3-pip python3-setuptools; else echo "Not Pycharm"; fi
 
-# Copy sh scripts
-COPY --chown=0:0 scripts/*.sh /opt/
+RUN true \
+    # Any command which returns non-zero exit code will cause this shell script to exit immediately.
+    && set -e \
+    # Activate debugging to show execution details: all commands will be printed before execution.
+    && set -x \
+    && adduser -r -u 1001 -G root -d $HOME -m -s /bin/sh $PROJECTOR_USER_NAME \
+    && echo "%wheel ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers \
+    && mkdir /projects \
+    && for f in "${HOME}" "/etc/passwd" "/etc/group /projects"; do\
+           sudo chgrp -R 0 ${f} && \
+           sudo chmod -R g+rwX ${f}; \
+       done \
+    && cat /etc/passwd | sed s#root:x.*#root:x:\${USER_ID}:\${GROUP_ID}::\${HOME}:/bin/bash#g > ${HOME}/passwd.template \
+    && cat /etc/group | sed s#root:x:0:#root:x:0:0,\${USER_ID}:#g > ${HOME}/group.template \
+    # Change permissions to allow editing of files for openshift user
+    && find $HOME -exec chgrp 0 {} \; -exec chmod g+rwX {} \;
 
-RUN mkdir -p /home/user && \
-    chgrp -R 0 /home && \
-    chmod -R g=u /etc/passwd /etc/group /home && \
-    chmod +x /opt/*.sh
+COPY --chown=$PROJECTOR_USER_NAME:root --from=projectorStaticFiles $PROJECTOR_DIR $PROJECTOR_DIR
 
-USER 10001
+RUN true \
+    # Any command which returns non-zero exit code will cause this shell script to exit immediately.
+    && set -e \
+    # Activate debugging to show execution details: all commands will be printed before execution.
+    && set -x \
+    && find $PROJECTOR_DIR -exec chgrp 0 {} \; -exec chmod g+rwX {} \; \
+    && find $PROJECTOR_DIR -name "*.sh" -exec chmod +x {} \; \
+    && mv $PROJECTOR_DIR/entrypoint.sh /entrypoint.sh \
+    && mv $PROJECTOR_DIR/$PROJECTOR_USER_NAME/.config $PROJECTOR_DIR/.default \
+    && rm -rf $PROJECTOR_DIR/$PROJECTOR_USER_NAME \
+    && cp -rp $PROJECTOR_DIR/.default $PROJECTOR_CONFIG_DIR \
+    && echo "idea.config.path=${PROJECTOR_CONFIG_DIR}/config" > $PROJECTOR_DIR/ide/bin/idea.properties \
```

Comment:
writing to a file could be handled by a shell script, so it's not so verbose here. See examples in https://github.com/eclipse-che/che-devfile-registry/tree/master/build/scripts

Reply:
Moved to the `entrypoint.sh`